integ/wait_until_search_attributes_test.go

```diff
@@ -0,0 +1,115 @@
+package integ
+
+import (
+	"context"
+	"fmt"
+	"github.com/indeedeng/iwf/service/common/ptr"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/indeedeng/iwf/gen/iwfidl"
+	"github.com/indeedeng/iwf/integ/workflow/wait_until_search_attributes"
+	"github.com/indeedeng/iwf/service"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWaitUntilSearchAttributesWorkflowTemporal(t *testing.T) {
+	if !*temporalIntegTest {
+		t.Skip()
+	}
+	for i := 0; i < *repeatIntegTest; i++ {
+		doTestWaitUntilSearchAttributes(t, service.BackendTypeTemporal, &iwfidl.WorkflowConfig{
+			ExecutingStateIdMode: ptr.Any(iwfidl.DISABLED),
+		})
+		smallWaitForFastTest()
+	}
+
+	for i := 0; i < *repeatIntegTest; i++ {
+		doTestWaitUntilSearchAttributes(t, service.BackendTypeTemporal, &iwfidl.WorkflowConfig{
+			ExecutingStateIdMode: ptr.Any(iwfidl.ENABLED_FOR_ALL),
+		})
+		smallWaitForFastTest()
+	}
+
+	// TODO: Rethink how this can be tested
+	// for i := 0; i < *repeatIntegTest; i++ {
+	// doTestWaitUntilSearchAttributes(t, service.BackendTypeTemporal, nil) // defaults to ExecutingStateIdMode: ENABLED_FOR_STATES_WITH_WAIT_UNTIL
+	// smallWaitForFastTest()
+	// }
```

Comment:
The last thing to do here is adding a test for the `skipWaitUntil` case

Reply:
👍 
Potentially we can just use a single workflow with two states to test it:
```
S1: skipWaitUntil, execute will just sleep for 5 seconds, then go to S2
S2: waitUntil for a signalA, execute will then go to S3
S3: waitUntil for a signalB, execute will then go to S3
```
S1 will have a 5 seconds window to assert for the SA.
S2/S3 can let us assert on the SA and then send the signal (and verify the previous one is gone)

This can be testing against different modes.